tests/dummy/app/templates/components/modal.hbs

```diff
@@ -4,6 +4,8 @@
   {{#uk-modal
     visible=visible
     on-submit=(action "submit")
+    on-show=(action (mut visible) true)
```

Comment:
`on-hide` makes sense to me because of DDAU, but `on-show` looks super unnecessary: The "hide" event can be triggered from the modal, but the "show" event can't. I'd say the default implementation for show should remain in the component, and `on-show` would then be an extremely optional hook that we'd probably need only for very special use cases. 

The public API would then be:
```hbs
{{#uk-modal
  visible=visible # component reacts to outside state of "visible"
  on-hide=(...)
}} 
```

Reply:
If I'm not mistaken, the modal can still be opened via a button. For that case we should have the possibility to update our toggle variable.

Reply:
Right. But let's not force the user to specify `on-show`, because with the documented usage it's redundant.